tests/providers/azure/azure_test.py

```diff
@@ -13,3 +13,410 @@
+MOCK_LIST_DISKS = {
+    'fake-disk-name': FAKE_DISK,
+    'fake-boot-disk-name': FAKE_BOOT_DISK
+}
+
+
+class TestAccount(unittest.TestCase):
+  """Test Azure account class."""
+  # pylint: disable=line-too-long
+
+  @mock.patch('libcloudforensics.providers.azure.internal.common.ExecuteRequest')
+  @typing.no_type_check
+  def testListInstances(self, mock_request):
+    """Test that instances of an account are correctly listed."""
+    mock_request.return_value = MOCK_REQUEST_INSTANCES
+
+    # If we don't specify a resource group name, the 'list_all' method should
+    # be called.
+    instances = FAKE_ACCOUNT.ListInstances()
+    mock_request.assert_called_with(mock.ANY, 'list_all')
+    self.assertEqual(1, len(instances))
+    self.assertIn('fake-vm-name', instances)
+    instance = instances['fake-vm-name']
+    self.assertEqual('fake-vm-name', instance.name)
+    self.assertEqual(
+        '/a/b/c/fake-resource-group/fake-vm-name', instance.resource_id)
+    self.assertEqual('fake-resource-group', instance.resource_group_name)
+    self.assertEqual('fake-region', instance.region)
+    self.assertEqual(['fake-zone'], instance.zones)
+
+    # If we specify a resource group name, the 'list' method should be called.
+    FAKE_ACCOUNT.ListInstances(
+        resource_group_name=instance.resource_group_name)
+    mock_request.assert_called_with(
+        mock.ANY, 'list', {'resource_group_name': instance.resource_group_name})
+
+  @mock.patch('libcloudforensics.providers.azure.internal.common.ExecuteRequest')
+  @typing.no_type_check
+  def testListDisks(self, mock_request):
+    """Test that disks of an account are correctly listed."""
+    mock_request.return_value = MOCK_REQUEST_DISKS
+
+    # If we don't specify a resource group name, the 'list' method should be
+    # called.
+    disks = FAKE_ACCOUNT.ListDisks()
+    mock_request.assert_called_with(mock.ANY, 'list')
+    self.assertEqual(2, len(disks))
+    self.assertIn('fake-disk-name', disks)
+    disk = disks['fake-disk-name']
+    self.assertEqual('fake-disk-name', disk.name)
+    self.assertEqual(
+        '/a/b/c/fake-resource-group/fake-disk-name', disk.resource_id)
+    self.assertEqual('fake-resource-group', disk.resource_group_name)
+    self.assertEqual('fake-region', disk.region)
+    self.assertEqual(['fake-zone'], disk.zones)
+
+    # If we specify a resource group name, the 'list_by_resource_group' method
+    # should be called.
+    FAKE_ACCOUNT.ListDisks(resource_group_name=disk.resource_group_name)
+    mock_request.assert_called_with(
+        mock.ANY,
+        'list_by_resource_group',
+        {'resource_group_name': disk.resource_group_name})
+
+  @mock.patch('libcloudforensics.providers.azure.internal.account.AZAccount.ListInstances')
+  @typing.no_type_check
+  def testGetInstance(self, mock_list_instances):
+    """Test that a particular instance from an account is retrieved."""
+    mock_list_instances.return_value = MOCK_LIST_INSTANCES
```

Comment:
No assert_called_with?

Reply:
On what, `mock_list_instances `? It only takes an optional parameter which is not used here. We could add an `assert_called_once` though, if we want. Wdyt?

Reply:
That sounds good!